### ISSUE TITLE
fix(nargs): a Java lambda with a bare-identifier parameter reports 0 args

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#1185]
## Summary

A Java lambda written with a bare-identifier parameter reports
`nargs = 0`:

```java
Function<Integer, Integer> f = x -> x + 1;   // nargs 0, should be 1
Function<Integer, Integer> g = (x) -> x + 1; // nargs 1
```

The two spellings are the same lambda — the parentheses are optional in
the grammar and carry no meaning — so this is the same
"byte-equivalent constructs score identically" contract the book states
for cognitive, broken on `nargs`.

## Cause

`compute_args` iterates the children of the `parameters` field. With
parentheses the lambda has a `formal_parameters` / `inferred_parameters`
node with one child; without them the parameter *is* a bare
`identifier`, which has no children, so the iteration yields nothing.

There is a singular-`parameter` fallback next to it that handles exactly
this shape — but it is gated to the JS family and never reached for
Java.

## Scope

Java is where it was measured. The bare-identifier lambda parameter is
not Java-specific — check Groovy and Kotlin, whose lambda/closure
grammars have the same optional-parentheses shape, before fixing only
Java. Per `.claude/rules/grammar-dispatch.md`, the fix for one language
here is the audit table for its siblings.

## Tests

- `x -> x + 1` and `(x) -> x + 1` must report the same `nargs`.
- A zero-parameter lambda `() -> 1` must stay 0 — the fallback must not
  count the body as a parameter.
- A multi-parameter lambda `(x, y) -> x + y` must stay 2, pinning that
  the fallback did not displace the plural path.
- `closure_nargs` vs `fn_nargs`: confirm the lambda still lands in the
  closure channel, not the function one.

Pre-existing on `main`; found while fixing #1160.
